accepted/2021/InboxSourceGenerators.md

```diff
@@ -0,0 +1,191 @@
+
+### Is it bad to have multiple versions of a source generator possibly loaded by a repository?
+
+A repository may have projects targeting multiple frameworks.  These frameworks may contain different versions of the same source generator.  When building the repository this may require the same compiler instance to load multiple copies of a source generator.  The same compiler would not load multiple copies of source generators which ship in a non-framework specific location.  This should work and be supported based on source generator versioning.
+
+### Should all inbox source generators be tied to a shared framework?
+
+No, some source generators might be independent of a target framework. One example of such a source generators may not contain functionality but may instead just build some convenience API on top of existing public API that is unlikely to ever change.  These could be considered stable enough to place in a non-framework-specific location like the SDK's analyzer folder.  Source generators which choose to ship this way should be highly specified and will need to support backwards compatibility and test on all frameworks.
+
+### Should only source generators be shipped this way?  What about analyzers and code-fixes?
+
+Nothing is preventing analyzers and code fixes from shipping in the same manner.  If it is desirable for a such a component to ship with the framework rather than the compiler then that component may ship in this manner.  We should only do this if we have a good reason, stable analyzers that are lanaguage extensions or tied to stable public API should ship non-framework-specific with the SDK as they do today.  Analyzers that need to validate source generator usage, or analyzers that are closely tied to runtime functionality may consider shipping with the framework.
+
+### Why in the ref pack, why not some other "analyzer" pack?
+
+The SDK already has a means for acquisition and selection of ref packs.  The inputs to the selection of ref packs are the same as that of source generators (Version, FrameworkReference, and TargetFramework), and the outputs are consumed by the same phase (compile) with the same level of cross-targeting granularity (TargetFramework and references).
+
+It may be desirable to create a separate package just for analyzers to reduce risk in servicing.  We support servicing ref packs today, but we must be careful not to expose new API nor change any assembly identities.  In my opinion it makes more sense to invest efforts in ensuring safe servicing of ref packs, with potential increased complexity in our own builds.  There is benefit in having servicable ref packs regardless and this keeps additional complexity (resolution of additional pack) out of user builds.
```

Comment:
```suggestion
It may be desirable to create a separate package just for analyzers to reduce risk in servicing.  We support servicing ref packs today, but we must be careful not to expose new API nor change any assembly identities.  It makes more sense to invest efforts in ensuring safe servicing of ref packs, with potential increased complexity in our own builds.  There is benefit in having serviceable ref packs regardless and this keeps additional complexity (resolution of additional pack) out of user builds.
```